README.md

```diff
@@ -404,8 +404,8 @@ These variables are set in the same level of the `logging_inputs`, `logging_outp
 - `logging_mark`: Mark message periodically by immark, if set to `true`. Default to `false`.
 - `logging_mark_interval`: Interval for `logging_mark` in seconds. Default to `3600`.
 - `logging_purge_confs`: `true` or `false`. If set to `true`, files in /etc/rsyslog.d are purged.
-- `logging_backup_dir`: Path to specify the directory to store backup file which contains pre-logging system role run. The backup is used when `logging_restore_confs` is set to `true`.
-- `logging_restore_confs`: `true` or `false`. If set to `true`, config files in the backup file are restored. This variable is to be used in the cleaning up.
+- `logging_backup_dir`: Path to specify the directory to store backup file which contains the config files installed by the other packages. Default to `/var/lib/rsyslog` for `rsyslog`.
```

Comment:
What other packages?

Reply:
Could you please elaborate your question?

If it helps, here's what's happening then.
Let's supposed `/etc/rsyslog.d` has these config files.
```
# ls /etc/rsyslog.d
00-global.conf                21-cloudinit.conf                   30-output-forwards-forwards_facility_only.conf
05-common-defaults.conf       30-output-files-files_output0.conf  30-output-forwards-forwards_severity_and_facility.conf
10-output-files-modules.conf  30-output-files-files_output1.conf  90-input-basics-basic_input0.conf
```
All files except `21-cloudinit.conf` do not belong to any rpm packages since they were generated by the logging role. The backup file in `logging_backup_dir` would contain just `21-cloudinit.conf` in this example and the file would be restored when purge is correctly executed (#240) or when `logging_reset_confs` is set to `true`.

Reply:
I am nitpicking the docs :) If you could specify exactly what packages override config files it would help. But it's up to you, if users does not need to know it, it is fine to keep this as is.

Reply:
This is how you find out which packages have files installed in `/etc/rsyslog.d`
```
# rpm -qf /etc/rsyslog.d
cloud-init-20.3-10.el8_4.2.noarch
rsyslog-8.1911.0-7.el8.x86_64
```
This is how you find out which files are owned by those packages in `/etc/rsyslog.d`
```
# for pkg in $(rpm -qf /etc/rsyslog.d); do echo "$pkg"; rpm -ql "$pkg" | grep \^/etc/rsyslog.d; done
cloud-init-20.3-10.el8_4.2.noarch
/etc/rsyslog.d
/etc/rsyslog.d/21-cloudinit.conf
rsyslog-8.1911.0-7.el8.x86_64
/etc/rsyslog.d
```

Reply:
There may be more - we can't assume that only cloud-init provides files in /etc/rsyslog.d - on some systems there is a listen.conf

Reply:
so the correct way to do this would be:
* rm -rf /etc/rsyslog.conf /etc/rsyslog.d/*
* find packages that provide files in /etc/rsyslog.d
* remove and reinstall those packages

Reply:
> so the correct way to do this would be:

At what timing, should these steps be executed? In this pr, it is done when `logging_reset_confs` is set to true and the code exists prior to the primary rsyslog role. Basically, it is "independent" from `logging_purge_confs`.

>     * rm -rf /etc/rsyslog.conf /etc/rsyslog.d/*
> 
>     * find packages that provide files in /etc/rsyslog.d
> 
>     * remove and reinstall those packages

In this pr, "find config files that belong to rpm packages except rsyslog and the config files are stored in `backup_additional.tgz`" at https://github.com/nhosoi/logging/blob/restore_confs/roles/rsyslog/tasks/main.yml#L81-L93. It's always done prior to the logging/rsyslog role execution. (This may be better to be done only when `backup_additional.tgz` does not exist.)

If `logging_reset_confs` is set to `true` and `backup_additional.tgz` exists, `rsyslog` is uninstalled and reinstalled, then the config files from `backup_additional.tgz` are extracted and placed in `/etc/rsyslog.d`. Please note this process does not clean up config files, e.g., config files generated by the previous logging role run. I guess you suggest to remove all of them by `rm -rf /etc/rsyslog.conf /etc/rsyslog.d/*`, right? > @richm. If yes, I'm updating this pr. Thanks.

Reply:
creating the backup.tgz if it does not exist will work fine only if you can guarantee that /etc/rsyslog.conf and the contents of the files in /etc/rsyslog.d are from a brand new system and have never been altered - also - what happens if you do the following:
```
install rsyslog
run logging role - creates backup first time
install cloud-init - installs /etc/rsyslog.d/21-cloudinit.conf
run logging role with purge/restore set - removes /etc/rsyslog.d/21-cloudinit.conf and restores from backup without /etc/rsyslog.d/21-cloudinit.conf
```
so you have a situation where a package is missing a file